frontend/src/pages/main/MainMaps.jsx

```diff
@@ -482,28 +430,63 @@ const MainMaps = ({ question, current }) => {
       url = generateAdvanceFilterURL(advanceSearchValue, url);
       api
         .get(url)
-        .then((res) => {
-          const { option } = shapeQuestion;
-          const { calculatedBy } = current.maps.shape;
-          let data = res.data;
-          if (shapeQuestion?.type === 'option' && calculatedBy) {
-            // fetch option value to calculated from question options
+        .then(({ data }) => {
+          const { scores, data: apiData } = data;
+          const { calculatedBy, id: shapeId } = current.maps.shape;
+          const { selectableMarkerDropdown } = current;
+          if (selectableMarkerDropdown?.length) {
+            const _markerOptions = selectableMarkerDropdown?.map((md) => {
+              const findMarker =
+                typeof md?.id === 'string'
+                  ? scores?.find((s) => s?.name === md?.id)
+                  : question?.find((q) => q?.id === md?.id);
+              const markOptions = findMarker?.option || findMarker?.labels;
+              return {
+                ...findMarker,
+                ...md,
+                option: markOptions,
+              };
+            });
+            setMarkerOptions(_markerOptions);
+
+            if (typeof selectableMarkerQuestion?.id === 'undefined') {
```

Comment:
Could you please explain why we need to check the type of `selectableMarkerQuestion?.id` ?

Reply:
after I looked again it turned out that this did not need to be checked